src/handleLearnCliCommand.ts

```diff
@@ -5,8 +5,64 @@ import {
 	getLatestCommitHash,
 	isFileInGitDirectory,
 } from './gitCommands.js';
-import { spawnSync } from 'node:child_process';
-import { dirname } from 'node:path';
+import { exec, spawnSync } from 'node:child_process';
+import { dirname, extname } from 'node:path';
+import { Project, SourceFile, SyntaxKind } from 'ts-morph';
+import * as tmp from 'tmp';
+import fs from 'node:fs';
+
+const isJSorTS = (name: string) =>
+	name.startsWith('.ts') || name.startsWith('.js');
+const variableDeclarationRegex =
+	/(?:export\s+)?(?:var|let|const|type)\s+([a-zA-Z_$][a-zA-Z0-9_$]*)/;
+
+const getFileExtension = (filePath: string) => {
+	return extname(filePath).toLowerCase();
+};
+
+const getOldSourceFile = (
+	commitHash: string,
+	filePath: string,
+	fileExtension: string,
+): Promise<SourceFile | null> => {
+	return new Promise((resolve, reject) => {
+		if (!isJSorTS(fileExtension)) {
```

Comment:
this check should be done out of a promise (before it)

Reply:
addressed